nexus/src/lib.rs

```diff
@@ -158,34 +174,15 @@ impl Server {
     /// immediately after calling `start()`, the program will block indefinitely
     /// or until something else initiates a graceful shutdown.
     pub async fn wait_for_finish(self) -> Result<(), String> {
-        let mut errors = vec![];
-        for server in self.http_servers_external {
-            errors.push(server.await.map_err(|e| format!("external: {}", e)));
-        }
-        errors.push(
-            self.http_server_internal
-                .await
-                .map_err(|e| format!("internal: {}", e)),
-        );
-        let errors = errors
-            .into_iter()
-            .filter(Result::is_err)
-            .map(|r| r.unwrap_err())
-            .collect::<Vec<String>>();
-
-        if errors.len() > 0 {
-            let msg = format!("errors shutting down: ({})", errors.join(", "));
-            Err(msg)
-        } else {
-            Ok(())
-        }
+        self.apictx.nexus.close_servers().await
```

Comment:
Tiny nit - it looks like this and `close()` are the same (except that `close()` unwraps and this doesn't). Would it make sense for `close()` to call this method so if it changes in the future only this one needs updating?

Reply:
Done in b9cfb6ea2d191d9bfaec4030e8d9412cc1232969